zipserver/readers.go

```
@@ -23,13 +23,13 @@ func annotatedReader(reader io.Reader) readerClosure {
 
 // wraps a reader to fail if it reads more than max of maxBytes, also tracks
 // the total amount of bytes read
-func limitedReader(reader io.Reader, maxBytes int, totalBytes *int) readerClosure {
+func limitedReader(reader io.Reader, maxBytes uint64, totalBytes *uint64) readerClosure {
 	remainingBytes := maxBytes
 	return func(p []byte) (int, error) {
 		bytesRead, err := reader.Read(p)
-		remainingBytes -= bytesRead
+		remainingBytes -= uint64(bytesRead)
 
-		*totalBytes += bytesRead
+		*totalBytes += uint64(bytesRead)
 
 		if remainingBytes < 0 {
 			return bytesRead, fmt.Errorf("File too large (max %d bytes)", maxBytes)
```

what happened to checking all limits upfront? :) this sounds like a "while-we're-already-deep-into-uploading stuff" check

---

discussed - @GranPC would rather have malformed zip files error out rather than have a silent failure & some corrupted files. fair enough!